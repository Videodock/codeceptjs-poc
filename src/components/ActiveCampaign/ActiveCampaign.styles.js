import { createStyles } from '@material-ui/styles';

export default ({ shape, palette, breakpoints }) => createStyles({
  author: {
    color: palette.text.blue,
  },
  section: {
    margin                  : 44,
    maxWidth                : 1440,
    [breakpoints.down('sm')]: {
      margin       : 16,
      marginBottom : 24,
      '& button, a': {
        width: '100%',
      },
    },
  },
  work: {
    display     : 'flex',
    minHeight   : 54,
    marginBottom: 14,
  },
  workLink: {
    '&:hover': {
      '& $workMusicIcon': {
        opacity: 0,
      },
      '& $workDownloadIcon': {
        opacity: 1,
      },
    },
    '& svg': {
      transition: 'opacity 0.2s ease',
      fill      : 'url(#FileMusicIconGradient)',
      height    : 36,
      width     : 30,
    },
    position      : 'relative',
    display       : 'flex',
    justifyContent: 'center',
    alignItems    : 'center',
    margin        : '8px 16px 8px 0',
    cursor        : 'pointer',
  },
  workMusicIcon: {
    opacity: 1,
  },
  workDownloadIcon: {
    opacity : 0,
    position: 'absolute',
  },
  actions: {
    '& a, button': {
      marginRight: 16,
    },
    marginBottom: 16,
  },
});
