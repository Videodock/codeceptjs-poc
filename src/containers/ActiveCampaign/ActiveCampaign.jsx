import React from 'react';
import PropTypes from 'prop-types';
import { gql } from 'apollo-boost';
import { useQuery } from '@apollo/react-hooks';
import ActiveCampaignComponent from '../../components/ActiveCampaign';
import NotFound from '../../screens/NotFound';

export default function ActiveCampaign({ campaignIdentifier }) {
  const { loading, error, data } = useQuery(GET_CAMPAIGN, { variables: { identifier: campaignIdentifier } });
  const campaign                 = data?.ControlAction[0];

  if (loading) {
    return null;
  }

  if (error) {
    return <h2>{error.message}</h2>;
  }

  if (!loading && !campaign) {
    return <NotFound />;
  }

  const digitalDocument = campaign.object.find(obj => obj.name === 'Work')?.nodeValue;

  return (
    <ActiveCampaignComponent
      campaignIdentifier={campaignIdentifier}
      author="TROMPA"
      title={campaign.name}
      description={campaign.description}
      scoreTitle={digitalDocument?.title}
      scoreSource={digitalDocument?.source}
    />
  );
}

ActiveCampaign.propTypes = {
  campaignIdentifier: PropTypes.string,
};

export const GET_CAMPAIGN = gql`
    query Campaign($identifier: ID!) {
        ControlAction (identifier: $identifier) {
            identifier
            name
            description
            object {
                ... on PropertyValue {
                    name
                    value
                    nodeValue {
                        ... on DigitalDocument {
                            identifier
                            title
                            source
                        }
                    }
                }
            }
        }
    }
`;
