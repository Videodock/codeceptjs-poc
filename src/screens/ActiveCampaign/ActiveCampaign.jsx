import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Helmet } from 'react-helmet';
import ShareIcon from '@material-ui/icons/Share';
import ShareDialog from '../../components/ShareDialog/ShareDialog';
import NavBar from '../../components/NavBar/NavBar';
import Jumbotron from '../../components/Jumbotron/Jumbotron';
import JumbotronContentCampaign from '../../components/JumbotronContentCampaign/JumbotronContentCampaign';
import ActiveCampaignProgress from '../../components/ActiveCampaignProgress/ActiveCampaignProgress';
import ActiveCampaignTwoSections from '../../components/ActiveCampaignTwoSections/ActiveCampaignTwoSections';

import images from '../../theme/images';

export default function ActiveCampaign ({ match }) {
  const { t }                                 = useTranslation('campaign');
  const { campaignIdentifier }                = match.params;
  const [shareDialogOpen, setShareDialogOpen] = useState(false);

  // temp static campaign data
  const campaignInfo = {
    campaignOwner: 'Annabel',
    campaignTitle: 'Summer concert: Mahler\'s 6th in the beer garden.',
    campaignUrl  : `https://trompamusic.eu/campaign/${campaignIdentifier}`,
    scoreTitle   : 'Mahler: Symphony No. 6 in A minor',
    scoreComment : 'Complete score',
  };

  return (
    <React.Fragment>
      <Helmet>
        <title>{t('jumbotron.primaryTitle')}</title>
        <meta name="description" content={t('meta_description')} />
      </Helmet>
      <NavBar
        navLinks={[{ name: t('navbar.share'), onClick: () => setShareDialogOpen(true), startIcon: <ShareIcon /> }]}
        primaryButton={{ name: t('navbar.join_campaign'), to: `/campaign/${campaignIdentifier}/task` }}
        drawerContent={<div />}
      />
      <ShareDialog
        open={shareDialogOpen}
        onClose={() => setShareDialogOpen(false)}
        modalContent={{
          title    : t('share_dialog.drum_up_support'),
          paragraph: t('share_dialog.lets_face_music'),
        }}
        campaignInfo={campaignInfo}
      />
      <Jumbotron
        image={images.mahlerSymphony}
        text={{
          prefixTitle          : t('jumbotron.prefixTitle'),
          primaryTitle         : t('jumbotron.primaryTitle'),
          secondaryTitle       : t('jumbotron.secondaryTitle'),
          introductionParagraph: t('jumbotron.introductionParagraph'),
        }}
        campaignInfo={campaignInfo}
        campaign
      >
        <JumbotronContentCampaign campaignInfo={campaignInfo} />
      </Jumbotron>
      <ActiveCampaignProgress />
      <ActiveCampaignTwoSections />
    </React.Fragment>
  );
}
