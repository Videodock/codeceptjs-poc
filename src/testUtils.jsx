import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
