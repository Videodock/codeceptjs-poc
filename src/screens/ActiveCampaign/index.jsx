import ActiveCampaign from './ActiveCampaign';

export * from './ActiveCampaign';
export default ActiveCampaign;
