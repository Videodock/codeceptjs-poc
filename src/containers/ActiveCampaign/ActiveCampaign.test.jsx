import React from 'react';
import { render } from '../../testUtils';
import ActiveCampaign from './ActiveCampaign';

describe('<ActiveCampaign />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ActiveCampaign />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
