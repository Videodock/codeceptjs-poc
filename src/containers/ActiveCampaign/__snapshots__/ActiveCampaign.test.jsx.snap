// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveCampaign /> renders and matches snapshot 1`] = `null`;
